pkg/kubectl/polymorphichelpers/mapbasedselectorforobject.go

```diff
@@ -50,57 +50,126 @@ func mapBasedSelectorForObject(object runtime.Object) (string, error) {
 		return generate.MakeLabels(t.Spec.Selector), nil
 
 	case *extensionsv1beta1.Deployment:
-		// TODO(madhusudancs): Make this smarter by admitting MatchExpressions with Equals
-		// operator, DoubleEquals operator and In operator with only one element in the set.
-		if len(t.Spec.Selector.MatchExpressions) > 0 {
-			return "", fmt.Errorf("couldn't convert expressions - \"%+v\" to map-based selector format", t.Spec.Selector.MatchExpressions)
+		var labels map[string]string
+		if t.Spec.Selector != nil {
```

Comment:
We could prob combine these two branches `if t.Spec.Selector != nil && len(t.Spec.Selector.MatchLabels) > 0 {`

Reply:
I don't think I can combine those two statements, since assigning to `labels` is different than returning the error.